libs/sdk-ui-pivot/src/CorePivotTable.tsx

```diff
@@ -738,64 +736,34 @@ export class CorePivotTablePure extends React.Component<ICorePivotTableProps, IC
         return columns.map((column: Column) => column.getColId());
     };
 
-    private getAutoResizedColumns = (columns: Column[]): IResizedColumns => {
-        return columns.reduce((acc, col) => {
-            const columnId = getColumnIdentifier(col);
-            const resizedColumn = acc[columnId];
-            if (resizedColumn) {
-                return acc;
-            }
-            return {
-                ...acc,
-                [columnId]: {
-                    width: col.getActualWidth(),
-                },
-            };
-        }, this.autoResizedColumns);
-    };
-
-    private autoresizeVisibleColumns = async (
-        columnApi: ColumnApi,
-        previouslyResizedColumnIds: string[],
-        firstCall: boolean = true,
-    ): Promise<void> => {
-        if (!this.shouldPerformAutoresize()) {
-            return Promise.resolve();
-        }
-
-        if (!this.isColumnAutoresizeEnabled()) {
-            return Promise.resolve();
-        }
+    private autoresizeColumnsByColumnId(columnApi: ColumnApi, columnIds: string[]) {
+        setColumnMaxWidth(columnApi, columnIds, AUTO_SIZED_MAX_WIDTH);
 
-        if (firstCall) {
-            await sleep(COLUMN_RESIZE_TIMEOUT);
-        }
+        columnApi.autoSizeColumns(columnIds);
 
-        const displayedVirtualColumns = columnApi.getAllDisplayedVirtualColumns();
-        const autoWidthColumnIds: string[] = this.getColumnIds(displayedVirtualColumns);
+        setColumnMaxWidth(columnApi, columnIds, MANUALLY_SIZED_MAX_WIDTH);
+    }
 
-        if (previouslyResizedColumnIds.length >= autoWidthColumnIds.length) {
-            this.autoResizedColumns = this.getAutoResizedColumns(columnApi.getAllDisplayedVirtualColumns());
+    private autoresizeAllColumns = async (gridApi: GridApi, columnApi: ColumnApi) => {
+        if (!this.shouldPerformAutoresize() || !this.isColumnAutoresizeEnabled()) {
             return Promise.resolve();
         }
 
-        const newColumnIds = difference(autoWidthColumnIds, previouslyResizedColumnIds);
-
-        this.autoresizeColumnsByColumnId(columnApi, newColumnIds);
-
         await sleep(COLUMN_RESIZE_TIMEOUT);
-        return this.autoresizeVisibleColumns(columnApi, autoWidthColumnIds, false);
+        const separators = get(this.props, ["config", "separators"], undefined);
+        const columns = columnApi.getPrimaryColumns();
+        const { headerFont, rowFont, subtotalFont, totalFont } = getTableFonts(this.containerRef);
+        this.autoResizedColumns = autoresizeAllColumns(columns, gridApi, columnApi, this.currentResult, {
```

Comment:
why retrieving columns from columnApi, then sending them both to the autoresize fun? perhaps enough to send the column api and let the autoresize the primary columns?

similar situation with the fonts. how about sending the containerRef and let the function do whatever it needs to do.

also i'm unsure about the separators. are you using this to estimate cell size? (data value + separators?) if so, what happens if the metric is customized with some totally custom format which adds bunch of stuff we don't expect?

Reply:
I little bit refactored autoresizeAllColumns.

About separators - separators are sent to this function https://github.com/gooddata/gdc-numberjs/blob/develop/lib/number.js#L543

After we have formatted text from numberjs function we measure that text. So if the metric is customized with some custom format that adds a bunch of stuff we just measure that custom format.

Reply:
oh right, I skimmed through the formatting. i wonder though, is it possible through ag-grid API get the formatted value?

Reply:
According to documentation: "You should use colDef.field most of the time. Use value getters when retrieving the data requires more logic, including executing your own expressions (similar to what a spreadsheet would do). " (https://www.ag-grid.com/javascript-grid-value-getters/) - I suppose that we do not an option to get the formatted value.